crates/threshold-bls/src/sig/tblind.rs

```diff
@@ -94,35 +115,79 @@ mod tests {
     #[test]
     fn tblind_g2_bellman_unblind() {
         aggregate_partially_unblinded::<G2Scheme<PCurve>>();
+        aggregate_partially_blinded::<G2Scheme<PCurve>>();
     }
 
     fn aggregate_partially_unblinded<B>()
     where
-        B: BlindThresholdScheme,
+        B: BlindThresholdScheme + SignatureScheme + ThresholdScheme,
     {
         let n = 5;
         let thr = 4;
         let (shares, public) = shares::<B>(n, thr);
         let msg = vec![1, 9, 6, 9];
 
         // blind the msg
-        let (token, blinded) = B::blind(&msg, &mut thread_rng());
+        let (token, blinded) = B::blind_msg(&msg, &mut thread_rng());
 
         // partially sign it
         let partials: Vec<_> = shares
             .iter()
-            .map(|share| B::partial_sign_without_hashing(share, &blinded).unwrap())
+            .map(|share| B::sign_blind_partial(share, &blinded).unwrap())
             .collect();
 
+        // verify if each blind partial signatures is correct
+        assert_eq!(
+            false,
+            partials
+                .iter()
+                .any(|p| B::verify_blind_partial(&public, &blinded, p).is_err())
+        );
+
         // unblind each partial sig
         let unblindeds: Vec<_> = partials
             .iter()
-            .map(|p| B::unblind_partial(&token, p).unwrap())
+            .map(|p| B::unblind_partial_sig(&token, p).unwrap())
             .collect();
 
         // aggregate
         let final_sig = B::aggregate(thr, &unblindeds).unwrap();
 
         B::verify(&public.public_key(), &msg, &final_sig).unwrap();
     }
+
+    fn aggregate_partially_blinded<B>()
```

Comment:
nit: To avoid code duplication, you could combine the two tests with a boolean `should_blind`, [in a similar way to how I've done it in the FFI/WASM tests](https://github.com/celo-org/celo-threshold-bls-rs/blob/master/crates/threshold-bls-ffi/src/ffi.rs#L836)

Reply:
Actually I could just combine the two so all tests are doing it.